jiant/modules/seq2seq_decoder.py

```diff
@@ -138,7 +189,16 @@ def forward(
     ) -> Dict[str, torch.Tensor]:
         # pylint: disable=arguments-differ
         """
-        Decoder logic for producing the entire target sequence at train time.
+        Forward pass for the decoder.
+
+        There are 3 possible modes:
+        1) Training: all metrics are computed given the last gold target tokens at each step
+        2) Validation [no training, but target tokens given]:
+               loss is computed given the gold target tokens;
+               accuracy/BLEU/etc. are computed using predicted tokens, no beam search
```

Comment:
isn't beam search being used (I guess with `k=1`, but the overall point being that we are also returning model generations in the validation setting) ?

Reply:
Yes, beam search is used during validation; but with beam 1